src/main/java/io/confluent/connect/jdbc/source/JdbcSourceTask.java

```diff
@@ -309,6 +313,43 @@ public void start(Map<String, String> properties) {
     maxRetriesPerQuerier = config.getInt(JdbcSourceConnectorConfig.QUERY_RETRIES_CONFIG);
   }
 
+  private void validateColumnsExist(
+      String mode, String incrementingColumn, List<String> timestampColumns, String table) {
+    try {
+      final Connection conn = cachedConnectionProvider.getConnection();
+      boolean autoCommit = conn.getAutoCommit();
+      try {
+        conn.setAutoCommit(true);
+        Map<ColumnId, ColumnDefinition> defnsById = dialect.describeColumns(conn, table, null);
+        Set<String> columnNames = defnsById.keySet().stream().map(ColumnId::name)
+            .map(String::toLowerCase).collect(Collectors.toSet());
+        if ((mode.equals(JdbcSourceTaskConfig.MODE_INCREMENTING)
+            || mode.equals(JdbcSourceTaskConfig.MODE_TIMESTAMP_INCREMENTING))
+            && !incrementingColumn.isEmpty()) {
+          if (!columnNames.contains(incrementingColumn.toLowerCase(Locale.getDefault()))) {
+            throw new ConnectException("Incrementing column " + incrementingColumn
+                + " does not exist");
+          }
+        }
+
+        if ((mode.equals(JdbcSourceTaskConfig.MODE_TIMESTAMP)
+            || mode.equals(JdbcSourceTaskConfig.MODE_TIMESTAMP_INCREMENTING))
+            && !timestampColumns.isEmpty()) {
+          for (String timestampColumn : timestampColumns) {
+            if (!columnNames.contains(timestampColumn.toLowerCase(Locale.getDefault()))) {
```

Comment:
We are returning at the first failure. Would it be better to check all column names at once and then throw an exception with all the missing columns.

You could form a `set` of timestamp columns as well and then find the missing ones as below.

```
        Set<String> lowercaseTsColumns = timestampColumns.stream()
            .map(column -> column.toLowerCase(Locale.getDefault()))
            .collect(Collectors.toSet());

        Set<String> missingTsColumns = lowercaseTsColumns.stream()
            .filter(column -> !columnNames.contains(column))
            .collect(Collectors.toSet());
```

Reply:
> Do you need add any new test cases for the method?
Yeah, I'll add one.

> Again wouldn't that be another security issue, that any malicious user can simply turn it off?
Oh yeah, I was thinking of an internal config but I forgot this is open source so it can be viewed.

Reply:
I realised we might lose the case sensitivity in column names if we convert to lowerCase, so better to do below instead.

```
        Set<String> missingTsColumns = timestampColumns.stream()
            .filter(column -> !columnNames.contains(column.toLowerCase(Locale.getDefault())))
            .collect(Collectors.toSet());
```